packages/core/src/render3/instructions/element.ts

```diff
@@ -43,24 +42,25 @@ import {registerInitialStylingOnTNode} from './styling';
  * @codeGenApi
  */
 export function ɵɵelementStart(
-    index: number, name: string, attrs?: TAttributes | null, localRefs?: string[] | null): void {
+    index: number, name: string, constsIndex?: number | null, localRefs?: string[] | null): void {
   const lView = getLView();
   const tView = lView[TVIEW];
+  const tViewConsts = tView.consts;
+  const consts = tViewConsts === null || constsIndex == null ? null : tViewConsts[constsIndex];
```

Comment:
Have we checked yet if there's any perf impact of the extra read using the micro-benchmarks?

Also, wouldn't `constIndex` be null if `tViewConsts` is null? Could we remove the `tViewConsts === null` check?

```suggestion
  const consts = constsIndex == null ? null : tViewConsts[constsIndex];
```

Reply:
I haven't run any if the benchmarks yet. Also the `constsIndex` could be null if we need to skip over it to pass in the `localRefs` param. I suppose we can get rid of it once the local refs are in the `consts` array.